### PR TITLE
MADmark improvements: new mechanism to load data and performance improvement of test case generation

### PR DESCRIPTION
**Pull request by huor**
This pull request have two big improvements:
1. Data loader tool refinement
   The new data loader tool is more flexible, which automatically download, convert, cache, and load dataset into testing databases.
   The new features include:
   (1) Each algorithm (Decision Tree, Random Forest, etc) has its own 'tables.yaml' to describe which dataset it needs and how they are generated (download origin data set from web, clean raw data, convert them to loadable data format, load them into database tables, etc).
   (2) Each algorithm has one or more converting tools to clean and convert related datasets. Users can reuse or modify them to meet their own requirement or add new tools themselves if neccessary.
   (3) Users can customize loading process. They can skip datasets, skip algorithms and append tables dynamically.  
      (3.1) Users can initialize testing database and load datasets for certain algorithms. 
      (3.2) Users can load all datasets except certain algorithms. 
      (3.3) Users can skip some big datasets if they want to load only some small/medium datasets and just do a smoke test.
      (3.4) Users can add database tables in append mode.
   (4) Besides permanent configuration files, Users can also temporarily customize loading process using parameter combinations in 'python run.py'.
   (5) All datasets are downloaded dynamically from website (UCI, etc). Thus MADmark is light-weighted, which contains only script and config files.
2. Performance improvement of test case generation
   Reduce the time of generating cases from 30min to about 10sec. By default, we remove the time-consuming logging part which takes about 30min. While the logging functionality is kept (use -d to enable it if necessary) as well.

### CODE REVIEW

**Comment by cwelton** [comment]
Ruilong, you and Jiali have pull permissions in the testsuite repository.  I recommend you review between the two of you and handle the pull on your own.  If you want additional review from development please engage Gavin.

**Comment by huor** [comment]
Thanks for your comment. We will review and merge it to master once Jiali and Gavin are back from vocation.

File: bin/run.py [hunk 1]
```diff
@@ -127,13 +133,25 @@ def main():
...
+
         for sqlfile in glob.glob('../testcase/*.sql'):
             utility.runSQL(psql_cmd, sqlfile, environ)
-    if options.genreport:
-        generateReport(psql_cmd, environ, options.genreport, 'all')
+
+    if options.smartload:
+        loading_manager = loadingManager('..', 'madlibtestdata')
+        loading_manager.do(options.module, False, False, False)
+#    if options.genreport:
```
Comment (huor) [inline bin/run.py]: Need to remove the generate report logic here.

File: bin/run.py [hunk 2]
```diff
@@ -4,18 +4,32 @@
 Most functions start with 'python run.py [-i][-g][-s]'"""
 import glob
 import os
-import sys
+import sys, subprocess
```
Comment (huor) [inline bin/run.py:7]: change
      "import sys, subprocess"
to
      "import sys"
      "import subprocess"

File: bootstrap/init.sql [hunk 3]
```diff
@@ -153,3 +76,327 @@ CREATE TABLE benchmark.evaluation_logistic_regression (
 	datasets text, 
 	coef double precision[]) DISTRIBUTED randomly;
 
+----decision_tree:
```
Comment (huor) [inline bootstrap/init.sql]: Remove R baseline result for DT.

File: doc/InstallGuide [hunk 4]
```diff
@@ -13,7 +13,7 @@ Step 02: Install database(s) for test cases execution purpose, multiple GPDBs/PG
                This host will also be used to install MADlib and MADMARK.
 
 Step 03: Download madmark and place the madmark source code in a folder, which will be $MADMARK_HOME
-         For example, download madmark at https://github.com/huor/testsuite.
+         For example, download madmark using 'p4 sync -f //analytics/madmark/main/...'
```
Comment (huor) [inline doc/InstallGuide]: Should be "For example, download madmark at https://github.com/madlib/testsuite"

**Pull request merged**